### DISCUSSION TITLE
Duplicated references no longer treated as a single PR

### DISCUSSION THREAD

**Discussion by CandiedCode** [renovatebot/renovate#22333] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.84.1-slim

### If you're self-hosting Renovate, select which platform you are using.

github.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I have a pyproject.toml that looks like something like this

```toml
[build-system]
requires = [
    "setuptools>=67.7.2",
    "wheel",
]

# https://packaging.python.org/en/latest/specifications/declaring-project-metadata/
[project]
name = "default_experiment_name"
version = "0.0.0"
description = "Please update Experiment name and description"
readme = "README.md"
requires-python = ">=3.8,<3.11"
keywords = []
classifiers = [
    # https://pypi.org/classifiers/
    "Private :: Do Not Upload",
    "Development Status :: 3 - Alpha",
    "Intended Audience :: Science/Research ",
    "Intended Audience :: Developers",
    "Natural Language :: English",
    "Operating System :: Unix",
    "Operating System :: POSIX",
    "Operating System :: Microsoft :: Windows",
    "Programming Language :: Python",
    "Programming Language :: Python :: 3",
    "Programming Language :: Python :: 3.8",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
]
dependencies = [
    "stainedglass==5.12.0",
]

[project.optional-dependencies]
cpu = [
    "stainedglass[cpu]==5.12.0",
]
gpu = [
    "stainedglass[gpu]==5.12.0",
]
test = [
    "stainedglass[test]==5.12.0",
]
pkg = [
    "stainedglass[pkg]==5.12.0",
]
```

In version 35.82.0, with my config file, It would onpen 1 PR but would update every line in pyproject.toml that referenced stainedglass.

branch name created example
branch=renovate/stainedglass-5.x

In version 35.84.1, it now wants to open seperate pr's for every reference for every optional dependency

branch names created example 
branch=renovate/cpu-stainedglass-5.x
branch=renovate/pkg-stainedglass-5.x

Is there a way to keep the original functionality so that all the versions for the same dependency stay in sync in 1 pr/

### What you tried so far.

Confirmed that behavior changed most likely in version [35.83.0](https://github.com/renovatebot/renovate/releases/tag/35.83.0)

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
@secustor wdyt?

**Comment by secustor** [comment]
The dependencies are not grouped because they are in different dependency groups. 

Is there a particular reason why you have the same dependency defined as a project one and then also multiple times in different groups?     
If you define it like this I would expect that you want different PRs for this. 

Also do you have logs from before [35.83.0](https://github.com/renovatebot/renovate/releases/tag/35.83.0)? My guess is there are some regexManagers used here.

**Reply by rarkins** [reply]
@secustor based on the screenshot below, it doesn't seem like a good idea to include the group name in the depName. What do you think about changing it in future?

**Reply by secustor** [reply]
I have no special feeling in both directions. 
It has been simply an assumption on my side. 

Tough I would change this in a major.

**Comment by CandiedCode** [comment]
@secustor  You are correct regexManager was picking this up before.

I looks like github dependabot updates all dependency groups btw
<img width="1536" alt="Screenshot 2023-09-18 at 11 55 21 AM" src="https://github.com/renovatebot/renovate/assets/8380706/ed274d9f-09fc-4b24-a48f-a6f1df7c1712">
